React/README.md

```diff
@@ -0,0 +1,10 @@
+# React
+
+1. [Intro to React](intro-to-react.md)
+1. [Hello World](react-hello-world.md)
+    - [Exercise](exercises/intro-to-react.md)
+1. [Creating Components](creating-components.md)
+1. [`props` & `state`](props-state.md)
+1. [Creating Components Part 2](creating-components-2.md)
+1. [Lifecycle Methods](lifecycle-methods.md)
+1. [Component Design](component-design.md)
```

Comment:
I know everything's in flux, but this order (Lifecycle Methods before Component Design) doesn't match the proposed calendar